src/mame/namco/namcos10.cpp

```diff
@@ -2494,23 +2660,80 @@ void namcos10_memp3_state::ns10_nicetsuk(machine_config &config)
 	namcos10_memp3_base(config);
 	namcos10_nand_k9f2808u0b(config, 8);
 
-	// NS10_TYPE2_DECRYPTER(config, m_decrypter, 0, logic);
+	subdevice<screen_device>("screen")->screen_vblank().set([this] (int state) {
+		// nicetsuk wants to start the pads and it hangs inside the pad callback routine
+		// due to issues with the PSX timer (rcnt) so disable the pad callbacks
+		// by nulling the addrs that would be used by PadStartCom.
+		// The actual game inputs are over JAMMA so perhaps his was left over from
+		// debug menus?
+		uint32_t *p_n_psxram = (uint32_t *) m_ram->pointer();
+
+		if( p_n_psxram[ 0x105e20 / 4 ] == 0x800de200 )
+		{
+			p_n_psxram[ 0x105e20 / 4 ] = 0;
+			p_n_psxram[ 0x105e24 / 4 ] = 0;
+		}
+	});
```

Comment:
This seems like a pretty nasty hack.  How broken is the timer emulation?  I realise MAME has always had issues with that.

Reply:
Using https://psx-spx.consoledev.net/timers/ as my source of information for reference.

I spent some time reading through the code but it gets stuck in an infinite loop due to the timer code, so I am going to say probably fairly broken. The timer control register is never initialized by the game so I don't know if it should have any defaults on a normal PS1. 

Here's my annotated Ghidra output for the broken code. I can provide the raw assembly routine if you'd prefer to read that.
```c
  if ((*0x1F801070 & 0x80) == 0) { // check I_STAT's IRQ7 Controller and Memory Card - Byte Received Interrupt
    do {
      if ((*0x1f801124 & 0x1000) == 0) { // check Reached FFFFh Value bit
        if ((*0x1f801124 & 0x808) == 0x808) { // check Reset counter to 0000h and Reached Target Value bits
          DAT_801acc1c = (uint)*0x1f801128 + DAT_801acc1c; // 0x1f801128 is counter target value register
        }
      }
      else {
        DAT_801acc1c = DAT_801acc1c + 0x10000;
      }
      if ((*0x1f801124 & 0x200) == 0) { // change comparison method based on clock source
        if (DAT_801acc18 << 3 <= (int)(((uint)*0x1f801120 + DAT_801acc1c) - last_saved_psx_timer_value)) { // x1f801120 is current timer value register
          return 0xfffffffe;
        }
      }
      else if (DAT_801acc18 <= (int)(((uint)*0x1f801120 + DAT_801acc1c) - last_saved_psx_timer_value)) {
        return 0xfffffffe;
      }
    } while ((*0x1F801070 & 0x80) == 0); // check I_STAT's IRQ7 Controller and Memory Card - Byte Received Interrupt
  }
```

It gets stuck in an infinite loop because the check `DAT_801acc18 << 3 <= (int)(((uint)*0x1f801120 + DAT_801acc1c) - last_saved_psx_timer_value)` never returns true. The worst I saw is that `last_saved_psx_timer_value` could end up a really high value like say 0xff40 (I saw something like 0xf640 or something really high, it depends on whatever timing the register was read at) and the trigger value can be 0x190 (a value I did see being used here) so you can end up such that it's never possible for the condition to be met.

Things that I see being an issue with the rcnt device are things like not pausing the timer during hblanks, reached target value not being emulated, and some other flags. 

Unless there's something I'm missing here which very well may be the case, I just don't think it's possible for this code to work in MAME.

0x800de9d8 is the function in question in the game's code if you or anyone else wants to take a look at it.

Reply:
> Unless there's something I'm missing here which very well may be the case, I just don't think it's possible for this code to work in MAME.

Oh, I’m not disputing that it won’t work in MAME with the current rcnt emulation.  It’s just that the rcnt emulation has been known to be broken since forever, and eventually it needs to be fixed.  We can’t put it off forever.  I really don’t like the idea of what amounts to an always-on cheat to get something into a “working” state.  It’s more honest to just mark the thing as not working.

Reply:
I can always turn this into a cheat and anyone who *really* wants to play it still can. Not that it's really playable right now anyway without the MP3 decoder.

Reply:
Look, I’d honestly much rather this potentially become the impetus needed for the rcnt to actually get fixed.  Gross hacks to paper over stuff that should have been fixed years ago are not the MAME way, and with the other high-quality PlayStation emulators around (i.e. Mednafen and DuckStation), we don’t have an excuse.

Reply:
Small win it is. Simply implementing bit 12 "Reached FFFFh Value" for the counter mode register fixed the issue.

Reply:
I pushed the commit. Please look it over and let me know if I missed anything obvious. I gave some other PSX-based games a quick go to make sure nothing obvious broke and it looked fine. I don't know what PSX-based games had hangs but it may be worth while to test any known ones to see if anything has improved?

https://github.com/mamedev/mame/pull/11131/commits/ee50e0a157e8a6fd7fc8393f654acd8f8866e139

Reply:
> Look, I’d honestly much rather this potentially become the impetus needed for the rcnt to actually get fixed.

There is no impetus required, it's in progress.